axelrod/strategies/axelrod_first.py

```diff
@@ -277,8 +406,8 @@ def strategy(self, opponent: Player) -> Action:
         if all(
             opponent.history[i] == self.history[i - 1]
             for i in range(1, len(self.history))
-        ):
-            # Check if opponent plays Tit for Tat
+        ) or opponent.history == self.history:
```

Comment:
Note that this is unlikely to be the original rule since strategies in the Fortran code would have to track their history themselves, and it's unlikely that the full history array was retained. I would guess that the original tracks cooperation and defection totals, the move on round 50, and the subsequent 5 moves, maybe with a bool "is_clone" or similar.

Of course we can't really know what it originally did so feel free to leave it as is.

Reply:
Very good spot, I'll add something about this to the docstring though.